### PR TITLE
Added new util class FirstOrderFilter

### PR DESCRIPTION
**Pull request by deanlee**
related to issue https://github.com/commaai/openpilot/issues/2462

### CODE REVIEW

File: selfdrive/ui/paint.cc [hunk 1]
```diff
@@ -223,11 +223,22 @@ static void ui_draw_vision_event(UIState *s) {
...
+#endif
 static void ui_draw_vision_face(UIState *s) {
+  static FirstOrderFilter filter(0., DISTRACTED_FILTER_TS, DT_DMON);
   const int face_size = 96;
   const int face_x = (s->viz_rect.x + face_size + (bdr_s * 2));
   const int face_y = (s->viz_rect.bottom() - footer_h + ((footer_h - face_size) / 2));
-  ui_draw_circle_image(s, face_x, face_y, face_size, "driver_face", s->scene.dmonitoring_state.getIsActiveMode());
+  const bool is_distracted = s->scene.dmonitoring_state.getIsDistracted();
+  const bool face_detected = s->scene.dmonitoring_state.getFaceDetected();
+  bool deactivate_icon = filter.update(!face_detected || is_distracted) > 0.64;
```
Comment (pd0wm) [inline selfdrive/ui/paint.cc:240]: Don't we need hysteresis? It could still flicker due to small changes around this value.

**Comment by deanlee** [comment]
revert ui_draw_vision_face, still use getIsActiveMode()

File: selfdrive/modeld/modeld.cc [hunk 2]
```diff
@@ -114,7 +111,7 @@ void run_model(ModelState &model, VisionIpcClient &vipc_client) {
 
       // tracked dropped frames
       uint32_t vipc_dropped_frames = extra.frame_id - last_vipc_frame_id - 1;
-      frames_dropped = (1. - frame_filter_k) * frames_dropped + frame_filter_k * (float)std::min(vipc_dropped_frames, 10U);
+      float frames_dropped = frame_dropped_filter.update((float)std::min(vipc_dropped_frames, 10U));
```
Comment (pd0wm) [inline selfdrive/modeld/modeld.cc:114]: The warmup is now different. You don't reset the internal filter state. You should probably add a `reset()` function to the filter for this.

File: selfdrive/modeld/modeld.cc [hunk 3]
```diff
@@ -111,8 +111,9 @@ void run_model(ModelState &model, VisionIpcClient &vipc_client) {
 
       // tracked dropped frames
       uint32_t vipc_dropped_frames = extra.frame_id - last_vipc_frame_id - 1;
+      if (run_count < 10) frame_dropped_filter.reset(0);  // let frame drops warm up
```
Comment (pd0wm) [inline selfdrive/modeld/modeld.cc:114]: Reset needs to be after the update, otherwise it could still trigger.

**Comment by pd0wm** [comment]
LGTM. Will do a quick test and merge!

**Pull request merged**